SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -563,10 +563,21 @@ void USpatialNetDriver::BeginDestroy()
 {
 	Super::BeginDestroy();
 
-	// If we are still connected, cleanup our corresponding worker entity if it exists.
+	// If we are still connected...
 	if (Connection != nullptr && WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID)
 	{
-		Connection->SendDeleteEntityRequest(WorkerEntityId);
+		// cleanup our corresponding worker entity if it exists.
+		if (WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID)
+		{
+			Connection->SendDeleteEntityRequest(WorkerEntityId);
+		}
+		
+		// destroy the connection to disconnect from SpatialOS if we aren't meant to persist it.
+		if (!bPersistSpatialConnection)
+		{
+			Connection->DestroyConnection();
```

Comment:
Should we instead call `FinishDestroy` to make sure the memory is cleaned up?

Reply:
No need to manually invoke GC. The connection object can be cleaned up normally.

Reply:
You never want to explicitly call `FinishDestroy` on UObjects. I believe the intention here is to stop the Spatial connection, and the actual instance doesn't have to be destroyed. That said, I would prefer that this code asks the game instance to destroy the connection and reset pointers to it.